CMakeLists.txt

```diff
@@ -6,11 +6,13 @@ if(NOT DEFINED CMAKE_MACOSX_RPATH)
   set(CMAKE_MACOSX_RPATH 0)
 endif()
 
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Commented in another PR

Reply:
cc: https://github.com/borglab/gtsam/pull/1392#discussion_r1081969554